app/services/container_dashboard_service.rb

```diff
@@ -0,0 +1,206 @@
+class ContainerDashboardService
+# Handle lack of metrics for each node?
+
+  def initialize(provider_id)
+    @provider_id = provider_id
+    @ems = ExtManagementSystem.find(@provider_id) unless @provider_id.nil?
+  end
+
+  def all_data
+    {
+      :status              => status,
+      :providers           => providers,
+      :node_utilization    => node_utilization,
+      :network_utilization => network_utilization,
+      :trends              => entity_trends,
+      :heatmaps            => heatmaps
+    }
+  end
+
+  def status
+    [
+      {
+        :name      => "Nodes",
+        :iconClass => "pficon-container-node",
+        :count     => @ems.present? ? @ems.container_nodes.count : ContainerNode.count,
+        :status    => [
+          {
+            :iconClass => "pficon-error-circle-o",
+            :count     => "3" #dummy data
+          }
+        ]
+      },
+      {
+        :name      => "Containers",
+        :iconClass => "fa fa-cube",
+        :count     => @ems.present? ? @ems.containers.count : Container.count
+      },
+      {
+        :name      => "Registries",
+        :iconClass => "pficon-registry",
+        :count     => @ems.present? ? @ems.container_image_registries.count : ContainerImageRegistry.count
+      },
+      {
+        :name      => "Projects",
+        :type      => "projects",
+        :iconClass => "pficon-project",
+        :count     => @ems.present? ? @ems.container_projects.count : ContainerProject.count
+      },
+      {
+        :name      => "Pods",
+        :iconClass => "fa fa-cubes",
+        :count     => @ems.present? ? @ems.container_groups.count : ContainerGroup.count,
+        :status    => [
+          {
+            :iconClass => "pficon-error-circle-o",
+            :count     => 3 #dummy data
+          }
+        ]
+      },
+      {
+        :name      => "Services",
+        :iconClass => "pficon-service",
+        :count     => @ems.present? ? @ems.container_services.count : ContainerService.count
+      },
+      {
+        :name      => "Images",
+        :iconClass => "pficon-image",
+        :count     => @ems.present? ? @ems.container_images.count : ContainerImage.count
+      },
+      {
+        :name      => "Routes",
+        :iconClass => "pficon-route",
+        :count     => @ems.present? && @ems.kind_of?(ManageIQ::Providers::Openshift::ContainerManager) ?
+                      @ems.container_routes.count :
+                      ContainerRoute.count
+      }
+    ]
+  end
+
+  def providers
+    # Change this for when @ems.present?
+    {
+      :title         => "Providers",
+      :count         => ManageIQ::Providers::Kubernetes::ContainerManager.count +
+                        ManageIQ::Providers::Openshift::ContainerManager.count,
```

Comment:
Is there a base ContainerManager you can query so you can avoid having to hardcode the types here?  This makes future pluggability difficult. cc @matthewd @blomquisg